### PR TITLE
Completely tears out all uses of the 'update_icon' var in mobs

### PR DESCRIPTION
**Pull request by Crazylemon64**
As stated above.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
This legit does nothing, doesn't it?

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 1]
```diff
@@ -9,7 +9,6 @@
 	var/next_attack = 0
 	var/pounce_cooldown = 0
 	var/pounce_cooldown_time = 30
-	update_icon = 1
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm]: i don't even know why this was here, aliens are hardcoded to call update_icons() on Life()

**Comment by Tastyfish** [comment]
I'm always in favor of reducing the ridiculous number for variables everything has.

File: code/game/dna/genes/goon_disabilities.dm [hunk 2]
```diff
@@ -348,8 +348,7 @@
 
 	L.adjust_fire_stacks(0.5)
 	L.visible_message("\red <b>[L.name]</b> suddenly bursts into flames!")
-	L.on_fire = 1
-	L.update_icon = 1
+	L.IgniteMob()
```
Comment (Fox-McCloud) [inline code/game/dna/genes/goon_disabilities.dm]: I knew there was a second instance of fire not properly updating your appearance.

**Pull request merged**